### PR TITLE
fix: Allow ticket owners to update and delete

### PR DESCRIPTION
**Pull request by swapnil-turing**

### CODE REVIEW

File: augment-store/server/ticket/tests.py [hunk 1]
```diff
@@ -307,11 +307,18 @@ def test_list_tickets_includes_created_at(self):
             self.assertIn("created_at", result)
             self.assertNotIn("is_deleted", result)
 
-    def test_update_ticket_forbidden_for_regular_user(self):
+    def test_update_ticket_owner_allowed(self):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:310]: value:useful; category:bug; feedback:Marked useful because it noticed the PR only exercised the update route even though delete shares the same view and access policy. The missing coverage could let owner delete or unrelated-ticket delete behavior diverge from the intended authorization rules without being caught.

File: augment-store/server/ticket/views.py [hunk 2]
```diff
@@ -112,7 +112,12 @@ class TicketDetailView(TicketBaseView, RetrieveAPIView):
 class TicketUpdateView(CacheInvalidatorMixin, TicketBaseView, RetrieveUpdateDestroyAPIView):
     serializer_class = TicketUpdateSerializer
     cache_service_class = TicketCacheService
-    permission_classes = [IsAuthenticated, hasAdminRole]
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/views.py]: value:useful; category:bug; feedback:Marked useful because it highlighted that expanding delete access made the endpoint's destructive behavior important to verify. It identified that a hard delete could permanently remove user tickets despite the model carrying an `is_deleted` flag and API expectations around soft deletion.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/ticket/views.py [hunk 3]
```diff
@@ -125,7 +125,8 @@ def perform_update(self, serializer):
 
     def perform_destroy(self, instance):
         reporter = instance.reporter
-        super().perform_destroy(instance)
+        instance.is_deleted = True
+        instance.save(update_fields=["is_deleted", "updated_at"])
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/views.py:129]: value:useful; category:bug; feedback:Marked useful because it caught a cache invalidation gap introduced by replacing the normal destroy path with a soft-delete write. The stale ticket cache could continue returning a ticket after deletion, which would make API responses disagree with the database state.

File: augment-store/server/ticket/views.py [hunk 4]
```diff
@@ -125,7 +125,8 @@ def perform_update(self, serializer):
 
     def perform_destroy(self, instance):
         reporter = instance.reporter
-        super().perform_destroy(instance)
+        instance.is_deleted = True
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/views.py:128]: value:useful; category:bug; feedback:Marked useful because it identified that soft deletion must be applied consistently across ticket lookup and aggregate paths. Without filtering deleted tickets everywhere, comment endpoints and stats could still expose or count tickets that users expect to be gone.

File: augment-store/server/ticket/tests.py [hunk 5]
```diff
@@ -370,10 +383,18 @@ def test_list_comments_excludes_is_deleted(self):
...
             self.assertIn("created_at", comment)
     
-    def test_delete_ticket_forbidden_for_regular_user(self):
+    def test_delete_ticket_owner_allowed(self):
         url = reverse("v1:ticket:delete_ticket", args=[self.ticket.id])
         response = self.authenticated_client.delete(url)
-        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+        self.ticket.refresh_from_db()
+        self.assertTrue(self.ticket.is_deleted)
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/tests.py:399]: value:good-to-have; category:bug; feedback:Marked good-to-have because it added focused coverage for a soft-delete edge case rather than catching the primary implementation defect. The risk was that comment list or create requests could still resolve a deleted ticket row and allow interaction with a ticket that should behave like missing.

**Comment by swapnil-turing** [comment]
augment review

File: augment-store/server/ticket/views.py [hunk 6]
```diff
@@ -112,7 +112,7 @@ class TicketDetailView(TicketBaseView, RetrieveAPIView):
 class TicketUpdateView(CacheInvalidatorMixin, TicketBaseView, RetrieveUpdateDestroyAPIView):
     serializer_class = TicketUpdateSerializer
     cache_service_class = TicketCacheService
-    permission_classes = [IsAuthenticated, hasAdminRole]
+    permission_classes = [IsAuthenticated]
```
[the remark this answers is not in the document]
Comment (swapnil-turing) [reply augment-store/server/ticket/views.py:115]: value:good-to-have; category:bug; feedback:Marked good-to-have because the comment called out an authorization ambiguity around the shared update/delete queryset. The risk was that assignee delete access could remain undocumented by tests, making a future policy change or regression hard to detect.

**Comment by swapnil-turing** [comment]
augment review

**Review by michaelos443 — APPROVED** [review]
